src/hooks/use-axios.js

```diff
@@ -0,0 +1,46 @@
+import { useState, useEffect, useCallback } from 'react'
+import axios from 'axios'
+
+const DELAY = 1 * 1000 // Artificial delay in ms
+
+function useAxios(run = false, urlRun, bodyRun, methodRun, headersRun) {
+  const [data, setData] = useState(null)
+  const [error, setError] = useState('')
+  const [loading, setLoading] = useState(false)
+
+  const callApiFn = useCallback(async (url, body, method, headers) => {
+    setLoading(true)
+    const options = body ? [url, body, { headers }] : [url, { headers }]
+    headers = headers || { 'content-type': 'application/json' }
+    method = method || (body ? 'post' : 'get')
+
+    try {
+      const res = await axios[method](...options)
+      const wait = (ms) => new Promise((res) => setTimeout(res, ms))
```

Comment:
remove this wait. If we need to add a wait later we can but it shouldn't interfere with the `set` calls in this hook. Basically what I'm worried about is correct handling of mount unmount etc.